### PR TITLE
Pay 1153 set up test case for tuesday using gregs spreadsheet

### PR DESCRIPTION
**Pull request by claus-humi**
Basic implementation of test cases for full QC calculation.

### CODE REVIEW

File: lib/taxman/taxman2023/factor.rb [hunk 1]
```diff
@@ -12,7 +12,11 @@ def initialize(**params)
     end
 
     def self.amount(context)
-      new(**context.slice(*params)).amount
+      parameters = params.each_with_object({}) do |param, values|
```
Comment (mark-h-humi) [inline lib/taxman/taxman2023/factor.rb:15]: why is this change necessary?

File: lib/taxman/taxman2023/factor.rb [hunk 2]
```diff
@@ -12,7 +12,11 @@ def initialize(**params)
...
+        values[param] = context.fetch(param)
+        values
```
Comment (mark-h-humi) [inline lib/taxman/taxman2023/factor.rb:17]: don't need this with `each_with_object`

File: lib/taxman/taxman2023/factor.rb (hunk 1 above)
Comment (claus-humi) [reply lib/taxman/taxman2023/factor.rb:15]: There was a bunch of things that were silently failing with similar nil errors to what you were using half a day on the other day. This gives us back the strictness that we had before hand with named parameters as slice does not complain if an element is missing.

So long story short, it makes debugging simpler and fails fast.

When we get around to the 2024 changes, this will make it easier to ensure that we can get them done quickly and not hunt small bugs for ages.

Or said in another way, it was faster for me to implement this than it was to hunt down all the issues we had with not passing in correct parameters.

File: lib/taxman/taxman2023/factor.rb (hunk 2 above)
Comment (claus-humi) [reply lib/taxman/taxman2023/factor.rb:17]: Nice! Started with reduce but the linter wanted each_with_object.

**Review by mark-h-humi — DISMISSED** [review]

**Review by mark-h-humi — DISMISSED** [review]

**Review by mark-h-humi — APPROVED** [review]
